integration-tests/smoke/keeper_test.go

```diff
@@ -585,6 +587,79 @@ func getKeeperSuite(
+					secondRegistry, _, _, _ := actions.DeployKeeperContracts(
+						registryVersion,
+						registryConfig,
+						defaultUpkeepsToDeploy,
+						defaultUpkeepGasLimit,
+						linkToken,
+						contractDeployer,
+						c,
+						linkFundsForEachUpkeep,
+					)
+
+					// Set the jobs for the second registry
+					actions.CreateKeeperJobs(chainlinkNodes, secondRegistry)
+					err = c.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Error creating keeper jobs")
+
+					err := registry.SetMigrationPermissions(common.HexToAddress(registry.Address()), 3)
+					Expect(err).ShouldNot(HaveOccurred(), "Couldn't set bidirectional permissions for first registry")
+					err = c.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to wait for migration permissions setting")
+
+					err = registry.SetMigrationPermissions(common.HexToAddress(secondRegistry.Address()), 3)
+					Expect(err).ShouldNot(HaveOccurred(), "Couldn't set bidirectional permissions for second registry")
+					err = c.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to wait for migration permissions setting")
+
+					var counterBeforeMigration *big.Int
+
+					// Check that the first upkeep from the first registry is performing (before being migrated)
+					Eventually(func(g Gomega) {
+						counterBeforeMigration, err = consumers[0].Counter(context.Background())
+						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+						g.Expect(counterBeforeMigration.Int64()).Should(BeNumerically(">", int64(0)),
+							"Expected consumer counter to be greater than 0, but got %s", counterBeforeMigration)
+					}, "1m", "1s").Should(Succeed())
+
+					upkeepInfo, err := registry.GetUpkeepInfo(context.Background(), upkeepIDs[0])
+					Expect(err).ShouldNot(HaveOccurred(), "Couldn't retrieve upkeep info")
+					err = c.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to wait for upkeep info retrieval")
+
+					// Grant permission to the registry to fund the upkeep
+					err = linkToken.Approve(registry.Address(), upkeepInfo.Balance)
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to approve")
+					err = c.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to wait for events")
+
+					// Migrate the upkeep with ID 0 from the first to the second registry
+					err = registry.Migrate([]*big.Int{upkeepIDs[0]}, common.HexToAddress(secondRegistry.Address()))
+					Expect(err).ShouldNot(HaveOccurred(), "Couldn't migrate the first upkeep")
+					err = c.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to wait for migration")
+
+					// Pause the first registry, in that way we make sure that the upkeep is being performed by the second one
+					err = registry.Pause()
+					Expect(err).ShouldNot(HaveOccurred(), "Could not pause the registry")
+					err = c.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to wait for the pausing of the first registry")
+
```

Comment:
can we refetch `counterAfterMigration` here and then check in eventually block that it increases?. It might have been performed more times in between all these txs

Reply:
Sure, updated it!